core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineMysql.java

```diff
@@ -235,9 +235,8 @@ public DataType getDataTypeForScalarDb(
       case CHAR:
       case VARCHAR:
       case LONGVARCHAR:
-        if (typeName.equalsIgnoreCase("ENUM")
-            || typeName.equalsIgnoreCase("SET")
-            || typeName.equalsIgnoreCase("JSON")) {
+        if (!typeName.toUpperCase().endsWith("CHAR") && !typeName.toUpperCase().endsWith("TEXT")) {
+          // to exclude ENUM, SET, JSON, etc.
```

Comment:
Changed to a more white-list-based check for the same reason in the SQL Server case below.